File: web-service/src/main/java/org/eclipse/model/Personne.java
package org.eclipse.model;

public abstract class Personne {
	/** Classe abstraite servent de base aux classes Client et Auteur */

	// Attributs
	private String nom;
	private String prenom;

	// Constructeurs
	public Personne() {
		super();
	}

	// Getters et Setters
	public Personne(String nom, String prenom) {
		super();
		this.nom = nom;
		this.prenom = prenom;
	}

	public String getNom() {
		return this.nom;
	}

	public void setNom(String nom) {
		this.nom = nom;
	}

	public String getPrenom() {
		return this.prenom;
	}

	public void setPrenom(String prenom) {
		this.prenom = prenom;
	}

	// toString
	@Override
	public String toString() {
		return "Personne [nom=" + this.nom + ", prenom=" + this.prenom + "]";
	}
}

File: angular-server/src/app/classes/model/personne.ts
export abstract class Personne {
    constructor(private _id: bigint = BigInt(0), private _civilite: string | null, private _nom: string, private _prenom: string | null) {}

    get id(): bigint {
        return this._id;
    }

    set id(_id: bigint) {
        this._id = _id;
    }

    get civilite(): string | null {
        return this._civilite;
    }
    
    set civilite(_civilite: string | null) {
        this._civilite = _civilite;
    }

    get nom(): string {
        return this._nom;
    }

    set nom(_nom: string) {
        this._nom = _nom;
    }

    get prenom(): string | null {
        return this._prenom;
    }

    set prenom(_prenom: string | null) {
        this._prenom = _prenom;
    }

}
